internal/core/data/v2/controller/http/event.go

```diff
@@ -54,7 +65,8 @@ func (ec *EventController) AddEvent(w http.ResponseWriter, r *http.Request) {
 	profileName := vars[v2.ProfileName]
 	deviceName := vars[v2.DeviceName]
 
-	addEventReqDTO, err := ec.reader.ReadAddEventRequest(r.Body)
+	reader := ec.getReader(r)
+	addEventReqDTO, err := reader.ReadAddEventRequest(r.Body)
```

Comment:
This implementation takes in a CBOR request and then publishes the request to the MessageBus as JSON. We are expecting it to publish the original CBOR to the MessageBus. V1, does this asynchronously. 
@cloudxxx8 , @jpwhitemn We should decide if V2 should also publish the CBOR asynchronous to  validation and persisting.

Reply:
@lenny-intel , I have added another commit, https://github.com/edgexfoundry/edgex-go/pull/3202/commits/a9e89425e353be16965c3e2b4be2414c6f565315, to update func PublishEvent to publish CBOR request to MessageBus in CBOR encoding byte array.  We could then open another issue for asynchronous PublishEvent once the community makes the final decision.